### ISSUE TITLE
Traeger integration uses `async_add_job` which is deprecated and will be removed in Home Assistant 2025.4

### ISSUE THREAD

**Issue by bakerkj** [njobrien1006/hass_traeger#71]
### System Health details

- Core: 2024.4.3
- Supervisor: 2024.04.0
- Operating System: 12.2
- Frontend: 20240404.1

### Checklist

- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/njobrien1006/hass_traeger/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Traeger integration uses `async_add_job` which is deprecated and will be removed in Home Assistant `2025.4`.

Details for fix can be found here:
https://developers.home-assistant.io/blog/2024/03/13/deprecate_add_run_job/

```
2024-04-12 10:17:18.377 WARNING (MainThread) [homeassistant.helpers.frame] Detected that custom integration 'traeger' calls `async_add_job`, which is deprecated and will be removed in Home Assistant 2025.4; Please review https://developers.home-assistant.io/blog/2024/03/13/deprecate_add_run_job for replaceme
nt options at custom_components/traeger/__init__.py, line 48: hass.async_add_job(, please create a bug report at https://github.com/njobrien1006/hass_traeger/issues
```

### Reproduction steps

1. Install Home Assistant 2024.4.3
2. Install Traeger integration 2024.03.29
3. Configure integration
4. Observe logs.

**Comment by bakerkj** [comment]
Possible fix in PR #72

**Comment by njobrien1006** [comment]
Likely fixed with #72. I just need to test in Hass O.S. and codespaces (straight Python ENV).
